bpf/lib/nodeport.h

```diff
@@ -461,9 +492,11 @@ static __always_inline int dsr_reply_icmp6(struct __ctx_buff *ctx,
 
 	wsum = ipv6_pseudohdr_checksum(&ip, IPPROTO_ICMPV6,
 				       bpf_ntohs(ip.payload_len), 0);
-	icmp.icmp6_cksum = csum_fold(csum_diff(NULL, 0, data + off, l3_max,
-					       csum_diff(NULL, 0, &icmp,
-							 sizeof(icmp), wsum)));
+	icmp.icmp6_cksum = csum_fold(csum_diff(NULL, 0, tmp, sizeof(tmp),
```

Comment:
Here you compute the checksum based on the original dgram, that one has the dest port pointing to the pod, right? Should not we have to set it to port port prior to start computing checksum? 

And I think we have the same kind of issue for ipv4

Reply:
Not really. I update the dport first on line number 482 (for ipv6) and 1584 (for ipv4) and then calculate the checksum. I verified the checksum correctness as this comment

https://github.com/cilium/cilium/pull/21234#issuecomment-1250005935

Reply:
Yes, my mistake, thank you.